modules/gapi/perf/common/gapi_video_perf_tests_inl.hpp

```diff
@@ -178,47 +178,53 @@ PERF_TEST_P_(BackgroundSubtractorPerfTest, TestPerformance)
         thr = 16.;
     else if (opType == gvideo::TYPE_BS_KNN)
         thr = 400.;
-    const gvideo::BackgroundSubtractorParams bsp(opType, histLength, thr, detectShadows, learningRate);
+    const gvideo::BackgroundSubtractorParams bsp(opType, histLength, thr, detectShadows,
+                                                 learningRate);
 
-    // Video source declaration
-    cv::VideoCapture cap;
-    if (!cap.open(findDataFile(filePath))) throw SkipTestException("Video file can not be opened");
-    cv::Mat frame;
-    cap >> frame;
+    // Retrieving frames
+    std::vector<cv::Mat> frames;
+    frames.reserve(testNumFrames);
+    {
+        cv::Mat frame;
+        cv::VideoCapture cap;
+        if (!cap.open(findDataFile(filePath)))
+            throw SkipTestException("Video file can not be opened");
+        for (std::size_t i = 0; i < testNumFrames && cap.read(frame); i++)
+        {
+            frames.push_back(frame);
+        }
+    }
+    testNumFrames = frames.size();
```

Comment:
Another change of count, it is curiously. testNumFrames = warm_up_frames + perf_test_work_frames, is it?

Reply:
This line is to avoid termination in case the video has less frames than it was required. So at this point `testNumFrames` is the real number of frames read

Reply:
This variable independent of test parameter. What if read() returns false (some problem) and we receive part of frames? Is it correct behavior for this case?

Reply:
Yes, exactly, this line handles the case when, for example, a video is too short and there are not enough frames. If so, I decided that the expected behaviour would be to process the obtainable amount of frames.

In the case when everything is all right the variable wouldn't change

Reply:
I hope "everything isn't right" case doesn't relate to this test and this step is just in case.

Reply:
What if frames.size() is zero?

Reply:
Yes, absolutely, this step is just in case.

If size is 0.. it's extremely unlikely, as video should have no frames despite the fact it have been read successfully.
But if it actually happened, nothing would be processed, so output Mats would remain empty. And then either validation would or would not pass. Which is not the expected behaviour

I'll add assert to have speakable error

Reply:
Assertion check must be here.

We can't have "flexible" scope for performance sampling. Test code must be fixed instead if error occurs.

Reply:
I see. Added termination if `frames.size()` is not `testNumFrames` after reading